fees/eastern-neighborhoods-non-residential-open-space/viewmodel.js

```diff
@@ -0,0 +1,76 @@
+    'json!./settings.json',
+    './component'
+], function(ko, AbstractFee, settings) {
+    var EasternNeighborhoodsNonResidentialOpenSpaceFee = function(params) {
+        this.settings = settings;
+
+        AbstractFee.apply(this, [params]);
+
+        this.triggered = ko.computed(function() {
+            return this.isProjectInArea() &&
+                (
+                    this.newNonRes() > 0 ||
+                    this.newNonRes()/this.existingNonResGFA() >= .2 ||
+                    this.nonResGFA()/this.existingNonResGFA() >= .2 ||
+                    this.officeGFA()/this.existingNonResGFA() >= .2
+
+                );
+        }, this);
+
+        this.ready = ko.computed(function() {
+            if (!this.triggered()) {
+                return true;
+            }
+            return this.retailGFA() !== null && this.retailGFA() !== '' &&
+                this.retailOpenSpaceProvidedGFA() !== null && this.retailOpenSpaceProvidedGFA() !== '' &&
+                this.manufacturingGFA() !== null && this.manufacturingGFA() !== '' &&
+                this.manufacturingOpenSpaceProvidedGFA() !== null && this.manufacturingOpenSpaceProvidedGFA() !== '' &&
+                this.officeGFA() !== null && this.officeGFA() !== '' &&
+                this.officeOpenSpaceProvidedGFA() !== null && this.officeOpenSpaceProvidedGFA() !== '' &&
+                this.otherGFA() !== null && this.otherGFA() !== '' &&
+                this.otherOpenSpaceProvidedGFA() !== null && this.otherOpenSpaceProvidedGFA() !== '';
+        }, this);
+
+
+        this.calculatedFee = ko.computed(function() {
+            if (!this.triggered()) {
+                return 0;
+            }
+
+            this.calculateRequiredOpenSpaceOrFee = function(a, b, c) {
+                return (a-b)*eval(c);
+            };
+
+            this.retailOpenSpaceRequiredGFA(
+                this.calculateRequiredOpenSpaceOrFee(this.retailGFA(), this.retailOpenSpaceProvidedGFA(), this.openSpaceRequirement['retail'])
+            );
+
+            this.manufacturingOpenSpaceRequiredGFA(
+                this.calculateRequiredOpenSpaceOrFee(this.manufacturingGFA(), this.manufacturingOpenSpaceProvidedGFA(), this.openSpaceRequirement['manufacturing'])
+            );
+
+            this.officeOpenSpaceRequiredGFA(
+                this.calculateRequiredOpenSpaceOrFee(this.officeGFA(), this.officeOpenSpaceProvidedGFA(), this.openSpaceRequirement['office'])
+            );
+
+            this.otherOpenSpaceRequiredGFA(
+                this.calculateRequiredOpenSpaceOrFee(this.otherGFA(), this.otherOpenSpaceProvidedGFA(), this.openSpaceRequirement['other'])
+            );
+
+            this.retailFee = (this.calculateRequiredOpenSpaceOrFee(this.retailOpenSpaceRequiredGFA(), this.retailOpenSpaceProvidedGFA(), this.feePerOpenSpaceGFA) > 0) ? this.calculateRequiredOpenSpaceOrFee(this.retailOpenSpaceRequiredGFA(), this.retailOpenSpaceProvidedGFA(), this.feePerOpenSpaceGFA) : 0;
```

Comment:
the same issue applies here as above.  Updating properties inside of a computed is not good practice.  Computeds should (in general) be pure functions. if these properties are not used elsewhere then they can just be variables local to the function.  Since these properties are used in the template, they should probably be their own computeds.